example/symmetric_difference.cpp

```diff
@@ -5,10 +5,14 @@
 #include <boost/hana/assert.hpp>
```

Comment:
Just like we did for the other functions, would it be possible to have an `example/set/symmetric_difference.cpp` and an `example/map/symmetric_difference.cpp` file? This would be consistent with the rest of the lib, too.